### PR TITLE
Release clangml-transforms.0.24

### PR DESCRIPTION
**Pull request by thierry-martinez**
New version compatible with Clangml 4

### CODE REVIEW

File: packages/clangml-transforms/clangml-transforms.0.24/opam [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+]
+maintainer: ["Thierry Martinez <thierry.martinez@inria.fr>"]
+authors: ["Thierry Martinez <thierry.martinez@inria.fr>"]
+bug-reports: "https://gitlab.inria.fr/memcad/clangml-transforms/issues"
+homepage: "https://gitlab.inria.fr/memcad/clangml-transforms"
+doc: "https://gitlab.inria.fr/memcad/clangml-transforms"
+license: "BSD"
+dev-repo: "git+https://gitlab.inria.fr/memcad/clangml-transforms"
+synopsis: "Code transformers for clangml"
+depends: [
```
Comment (kit-ty-kate) [inline packages/clangml-transforms/clangml-transforms.0.24/opam]:
```suggestion
depends: [
  "ocaml"
  "dune" {>= "1.10"}
```

File: packages/clangml-transforms/clangml-transforms.0.24/opam [hunk 2]
```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["dune" "build" "-p" name "@doc"] {with-doc}
```
Comment (kit-ty-kate) [inline packages/clangml-transforms/clangml-transforms.0.24/opam]:
```suggestion
  ["dune" "build" "-p" name "@doc" "-j" jobs] {with-doc}
```

**Comment by thierry-martinez** [comment]
@kit-ty-kate Thanks for your suggestions for the opam file. This opam file is generated by `dune` from the `dune-project` file: do you think it worth that I make a feature-request for `dune` to change how the opam file is generated (especially for the `"-j"` option with `with-doc`, but a lower-bound for `dune` version could be systematically added as well)?

**Comment by kit-ty-kate** [comment]
I think it might be worth doing it yes: Everything regarding opam files generation are here I think: https://github.com/ocaml/dune/blob/master/src/dune/opam_create.ml

Do you want me to do it or are you motivated to do it?

**Comment by kit-ty-kate** [comment]
oh, looking at the code, I think everything that you want (except `"-j" jobs`) is already done. But for some reason they require people to require at least dune 1.11 (marked in their dune-project file)

**Comment by kit-ty-kate** [comment]
ok so the version requirement is here on purpose and the output shouldn't be changed so it's not worth spending time looking into it

**Comment by thierry-martinez** [comment]
> ok so the version requirement is here on purpose and the output shouldn't be changed so it's not worth spending time looking into it

Thank you very much for having checked that: I will mark `dune-project` files with `1.11` from now on. Where did you find the purpose of the version requirement? (I was unable to find the explanation related to the commit https://github.com/ocaml/dune/commit/284dc0ba20c4bc9b036aab2c590f9a1bf43da898 ) Thanks again!

**Comment by thierry-martinez** [comment]
The compiler failure on `memcad` seems to be due to the uninstallability of older versions of `clangml-transforms` which rely on old versions of LLVM that are no longer packaged. A new version of `memcad` should be released soon, which will use the new `clangml-transforms`.

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Comment by kit-ty-kate** [comment]
> Where did you find the purpose of the version requirement? (I was unable to find the explanation related to the commit [ocaml/dune@284dc0b](https://github.com/ocaml/dune/commit/284dc0ba20c4bc9b036aab2c590f9a1bf43da898) ) Thanks again!

I asked the authors ^^"

**Pull request merged**